lxd/lifecycle/cluster.go

```diff
@@ -0,0 +1,33 @@
+package lifecycle
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/lxc/lxd/shared/api"
+)
+
+// ClusterAction represents a lifecycle event action for clusters.
+type ClusterAction string
+
+// All supported lifecycle events for clusters.
+const (
+	ClusterEnabled            = ClusterAction("enabled")
+	ClusterJoined             = ClusterAction("joined")
```

Comment:
So at the cluster level, I think we want:
 - Enabled (should be emitted when we turn the initial standalone server into the initial cluster member)
 - Disabled (should be emitted on a leaver cluster member)
 - TokenCreated (when a new token is created)
 - CertificateUpdated (when the certificate is updated)

At the member level we should have:
 - Added (replaces ClusterJoined)
 - Updated
 - Renamed
 - Removed (replaces Deleted)